docs/dns-health-checks.md

```diff
@@ -0,0 +1,86 @@
+# DNS Health Checks
+DNS Health Checks are a crucial tool for ensuring the availability and reliability of your multi-cluster applications. Kuadrant offers a powerful feature known as DNSPolicy, which allows you to configure and verify health checks for DNS endpoints. This guide provides a comprehensive overview of how to set up, utilize, and understand DNS health checks.
+
+## What are DNS Health Checks?
+DNS Health Checks are a way to assess the availability and health of DNS endpoints associated with your applications. These checks involve sending periodic requests to the specified endpoints to determine their responsiveness and health status. by configuring these checks via the [DNSPolicy](./dns-policy.md), you can ensure that your applications are correctly registered, operational, and serving traffic as expected.
+
+## Configuration of Health Checks
+To configure a DNS health check, you need to specify the `healthCheck` section of the DNSPolicy. The key part of this configuration is the `healthCheck` section, which includes important properties such as:
+
+* `allowInsecureCertificates`: Added for development environments, allows health probes to not fail when finding an invalid (e.g. self-signed) certificate.
+* `additionalHeadersRef`: This refers to a secret that holds extra headers, often containing important elements like authentication tokens.
+* `endpoint`: This is the path where the health checks take place, usually represented as '/healthz' or something similar.
+* `expectedResponses`: This setting lets you specify the expected HTTP response codes. If you don't set this, the default values assumed are 200 and 201.
+* `failureThreshold`: It's the number of times the health check can fail for the endpoint before it's marked as unhealthy.
+* `port`: Specific port for the connection to be checked.
+* `protocol`: Type of protocol being used, like HTTP or HTTPS. **(Required)**
+
+
+```
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: prod-web
+  namespace: multi-cluster-gateways
+spec:
+  targetRef:
+    name: prod-web
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  healthCheck:
+    allowInsecureCertificates: true
+    endpoint: /
+    expectedResponses:
+      - 200
+      - 201
+      - 301
+    failureThreshold: 5
+    port: 443
+    protocol: https
+EOF
```

Comment:
hmm is it not possible to specify how often it checks @philbrookes @sergioifg94 ? @Ygnas we should perhaps convey what the default polling period is in this doc also

Reply:
I think at the moment it's possible to specify it in the DNSHealthCheckProbes but not in the DNSPolicy. It'd be possible to add it as another field in the policy that gets propagated to the probes

Reply:
yeah ok I will add an issue for that, I think being able to control that is useful WDYT?

Reply:
https://github.com/Kuadrant/multicluster-gateway-controller/issues/472